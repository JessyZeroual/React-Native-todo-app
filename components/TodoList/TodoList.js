import React from "react";
import { FlatList, Text } from "react-native";
import { TodoListWrapper } from "./TodoList.styled";

const TodoList = ({ todos }) => {
  const renderItem = ({ item }) => <Text>{item.title}</Text>;
  return (
    <TodoListWrapper>
      <FlatList
        data={todos}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
      />
    </TodoListWrapper>
  );
};

export default TodoList;
