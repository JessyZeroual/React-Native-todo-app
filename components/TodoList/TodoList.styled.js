import styled from "styled-components/native";
import { primaryColor } from "../../constant/style-constants";

export const TodoListWrapper = styled.View`
  background: #ebebef;
  padding: 0px 10px;
  width: 100%;
`;
