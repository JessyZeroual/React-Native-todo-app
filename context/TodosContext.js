import React from "react";

const TodosContext = React.createContext({});

export const TodosContextProvider = TodosContext.Provider;
export const TodosContextConsumer = TodosContext.Consumer;
export default TodosContext;
