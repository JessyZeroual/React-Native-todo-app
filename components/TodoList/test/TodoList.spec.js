import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { TodosContextProvider } from "../../../context/TodosContext";
import { FlatList } from "react-native";
import TodoList from "../TodoList";

jest.mock("../../TodoItem/TodoItem", () => () => "View");

const todos = [
  {
    id: "id-1",
    name: "Todo-1",
  },
  {
    id: "id-2",
    name: "Todo-2",
  },
  {
    id: "id-3",
    name: "Todo-3",
  },
];

const fetchTodos = jest.fn(() =>
  todos.concat([
    {
      id: "id-4",
      name: "Todo-4",
    },
    {
      id: "id-5",
      name: "Todo-5",
    },
    {
      id: "id-6",
      name: "Todo-6",
    },
  ])
);

describe("TodoList", () => {
  let renderer;

  beforeEach(() => {
    act(() => {
      renderer = TestRenderer.create(
        <TodosContextProvider
          value={{
            state: { todos },
            dispatch: { fetchTodos },
          }}
        >
          <TodoList />
        </TodosContextProvider>
      );
    });
  });

  describe("When TodoList is mounted", () => {
    it("renders correctly", () => {
      expect(renderer).toMatchSnapshot();
    });

    it("receives a list of todos", () => {
      expect(renderer.root.findByType(FlatList).props.data.length).toEqual(3);
    });
  });

  describe("When list on end reached", () => {
    it("Fetch more todos", () => {
      act(() => {
        renderer.update(
          <TodosContextProvider
            value={{
              state: {
                todos: renderer.root.findByType(FlatList).props.onEndReached(),
              },
              dispatch: { fetchTodos },
            }}
          >
            <TodoList />
          </TodosContextProvider>
        );
      });

      expect(fetchTodos).toHaveBeenCalledTimes(1);
      expect(fetchTodos).toHaveBeenCalledWith({ needMoreTodos: true });
      expect(renderer.root.findByType(FlatList).props.data.length).toEqual(6);
    });
  });
});
