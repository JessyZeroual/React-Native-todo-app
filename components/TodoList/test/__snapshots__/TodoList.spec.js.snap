// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList When TodoList is mounted renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ebebef",
        "width": "100%",
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "id-1",
          "name": "Todo-1",
        },
        Object {
          "id": "id-2",
          "name": "Todo-2",
        },
        Object {
          "id": "id-3",
          "name": "Todo-3",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
